app/services/account_creation/device_profiling.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module AccountCreation
+  class DeviceProfiling
+    attr_reader :request_ip,
+                :threatmetrix_session_id,
+                :user_email,
+                :current_sp,
+                :device_profile_result
+    def proof(
+      request_ip:,
+      threatmetrix_session_id:,
+      user_email:,
+      current_sp: nil
+    )
+      @request_ip = request_ip
+      @threatmetrix_session_id = threatmetrix_session_id
+      @user_email = user_email
+      @current_sp = current_sp
+
+      @device_profile_result = device_profile
+    end
+
+    def device_profile
+      return threatmetrix_disabled_result unless
+        FeatureManagement.account_creation_device_profiling_collecting_enabled?
+      return threatmetrix_id_missing_result if threatmetrix_session_id.blank?
+      ddp_params = {}
+      ddp_params[:threatmetrix_session_id] = threatmetrix_session_id
+      ddp_params[:email] = user_email
+      ddp_params[:request_ip] = request_ip
+
+      proofer.proof(ddp_params)
```

Comment:
seems like this `ddp_params` hash cold be entirely inlined?
```suggestion
      proofer.proof(
        threatmetrix_session_id: threatmetrix_session_id,
        email: user_email,
        request_ip: request_ip,
      )
```